src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    let temporalState = { ...currentState };
+
+    if (action.type === 'addProperties' && action.extraData) {
+      for (const key in action.extraData) {
+        temporalState[key] = action.extraData[key];
```

Comment:
I don't understand why you need to use  logical and (&&) here
` if (action.type === 'addProperties' && action.extraData) {`

Reply:
and also maybe I wrong but in check list, need to use switch
[CODE STYLE]: Use switch statement if you have limited amount of conditions.
[CODE STYLE]: switch/case should always have default case for error handling.

Reply:
Yeah, you're right. I didn't look at the checklist. Thanks.